weather/location.go

```diff
@@ -16,12 +14,13 @@ type Location struct {
 	State     string  `json:"state"`
 }
 
-func (s *LocationService) FetchLatLonForCity(city string) (*Location, error) {
+func (s *WeatherClient) FetchLatLonForCity(city string) (*Location, error) {
 	var locations []Location
 
-	url := locationURL + s.client.apiKey + "&q=" + city
+	url := locationURL + s.apiKey + "&q=" + city
```

Comment:
Since all actions with the WeatherClient will need the apiKey, moving that logic to somewhere common would be a good idea. You could add it to the `Fetch` function and append it to the url there.